receipt/src/main/java/com/hyperwallet/android/receipt/view/ListReceiptFragment.java

```diff
@@ -0,0 +1,307 @@
+            @Override
+            public void onChanged(PagedList<Receipt> transferMethods) {
+                mListReceiptAdapter.submitList(transferMethods);
+            }
+        });
+
+        mListReceiptViewModel.getReceiptErrors().observe(getViewLifecycleOwner(),
+                new Observer<Event<List<HyperwalletError>>>() {
+            @Override
+            public void onChanged(Event<List<HyperwalletError>> listEvent) {
+                if (!listEvent.isContentConsumed()) {
+                    mOnLoadReceiptErrorCallback.showErrorOnLoadReceipt(listEvent.getContent());
+                }
+            }
+        });
+
+        mListReceiptViewModel.isLoadingData().observe(getViewLifecycleOwner(), new Observer<Boolean>() {
+            @Override
+            public void onChanged(Boolean loading) {
+                if (loading) {
+                    mProgressBar.setVisibility(View.VISIBLE);
+                } else {
+                    mProgressBar.setVisibility(View.GONE);
+                }
+            }
+        });
+    }
+
+    void retry() {
+        mListReceiptViewModel.retryLoadReceipts();
+    }
+
+    interface OnLoadReceiptErrorCallback {
+
+        void showErrorOnLoadReceipt(@NonNull final List<HyperwalletError> errors);
+    }
+
+    private static class ListReceiptItemDiffCallback extends DiffUtil.ItemCallback<Receipt> {
+
+        @Override
+        public boolean areItemsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+
+        @Override
+        public boolean areContentsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+    }
+
+    private static class ListReceiptAdapter
+            extends PagedListAdapter<Receipt, ListReceiptAdapter.ReceiptViewHolder> {
+
+        static final int HEADER_VIEW_TYPE = 1;
+        static final int DATA_VIEW_TYPE = 0;
+
+        ListReceiptAdapter(@NonNull final DiffUtil.ItemCallback<Receipt> diffCallback) {
+            super(diffCallback);
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            if (position != 0) {
+                Receipt previous = getItem(position - 1);
+                Receipt current = getItem(position);
+                if (isDataViewType(previous, current)) {
+                    return DATA_VIEW_TYPE;
+                }
+            }
+            return HEADER_VIEW_TYPE;
+        }
+
+        boolean isDataViewType(@NonNull final Receipt previous, @NonNull final Receipt current) {
+            Calendar prev = Calendar.getInstance();
+            prev.setTime(DateUtils.fromDateTimeString(previous.getCreatedOn()));
```

Comment:
@peter-joseph could we avoid parsing String in a Adapter, it's quite time-consuming operation. 
I would prefer holding Date in a Receipt. This will save time twice.

Reply:
Yep I like that approach actually it will save us time, what do you think @azakrevska-epam

Reply:
In second thought, we cannot save date instance to Receipt as well since createdOn in Receipt object is GMT, and in presentation layer we need to change it on whatever locale is the mobile user is using.

This code is as good as it can be